addons/mail/models/mail_link_preview.py

```diff
@@ -38,19 +39,28 @@ def _create_from_message_and_notify(self, message):
             'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; rv:91.0) Gecko/20100101 Firefox/91.0'
         })
         link_preview_values = []
+        link_preview_ids = []
+        link_previews_by_url = dict((preview.source_url, preview) for preview in message.link_preview_ids)
         for url in list(urls):
+            if url in link_previews_by_url:
+                preview = link_previews_by_url.pop(url)
+                if not preview.hide:
+                    link_preview_ids.append(preview.id)
+                continue
             if preview := link_preview.get_link_preview_from_url(url, requests_session):
                 preview['message_id'] = message.id
                 link_preview_values.append(preview)
-            if len(link_preview_values) > 5:
+            if len(link_preview_values) + len(link_preview_ids) > 5:
                 break
-        if link_preview_values:
-            link_previews = self.env['mail.link.preview'].create(link_preview_values)
+        for unused_preview in link_previews_by_url.values():
+            unused_preview._hide_and_notify()
+            unused_preview.unlink()
```

Comment:
I think we should make a different method for hide and unlink notify.

Reply:
but on the client side, we only care about `delete`, so I think maybe it is not needed to have both. On the server side, we have `_hide()` and `unlink()`. On the client side, we just need `mail.link.preview/delete` and no `mail.link.preview/hide`

Reply:
But it doesn't make sense to "first hide" and then "delete" in consecutive lines.
Maybe the bus stuff can be the same, in particular both should just be record/insert or record/delete